pages/verify/[discordServerId]/[discordMemberId]/[customLink].tsx

```diff
@@ -77,6 +113,29 @@ const VerifyPage = ({
     )
       setWrongStarknetNetwork(true);
     else setAccount(wallet.account);
+
+    const isArgentWallet = wallet.id.toLowerCase().includes("argent");
+    setIsArgent(isArgentWallet);
+
+    const currentChainId =
+      wallet.account?.provider.chainId ||
+      wallet.provider?.chainId ||
+      wallet.chainId;
+    const validChainIds = chainAliasByNetwork[starknetNetwork];
+
+    if (!validChainIds.includes(currentChainId)) {
+      setWrongStarknetNetwork(true);
+
+      if (isArgentWallet) {
+        await handleNetworkSwitch(wallet);
+        return;
+      }
+
+      return;
+    }
+
+    setAccount(wallet.account);
+    setWrongStarknetNetwork(false);
   }, [starknetNetwork]);
```

Comment:
You need to add `handleNetworkSwitch` there.
To avoid an unwanted rerenders, you'll need to use a UseCallback to define handleNetworkSwitch